Objective-C/Internal/CBLDocument+Internal.h

```diff
@@ -76,7 +78,7 @@ NS_ASSUME_NONNULL_BEGIN
                        documentID: (NSString*)documentID
                              body: (nullable FLDict)body;
 
-- (instancetype) initWithDatabase: (CBLDatabase*)database
+- (instancetype) initWithDatabase: (nullable CBLDatabase*)database
```

Comment:
I prefer to have a dedicate init for the remote doc such as 

```
- (instancetype) initRemoteDocumentWithID: (NSString*)documentID revisionID: (NSString*)revisionID body: (nullable FLDict)body;
```

In its implementation, you can add a flag such as `isRemoteDoc` to indicate that this document is a remote document, instead of checking whether database is nil for the remote doc. Basically, trying to be more specific.